sdk/Pulumi.Automation/Workspace.cs

```diff
@@ -123,6 +123,25 @@ internal Workspace(IPulumiCmd cmd)
         /// <param name="cancellationToken">A cancellation token.</param>
         public abstract Task PostCommandCallbackAsync(string stackName, CancellationToken cancellationToken = default);
 
+        /// <summary>
+        /// Adds environments to the end of a stack's import list. Imported environments are merged in order
+        /// per the ESC merge rules. The list of environments behaves as if it were the import list in an anonymous
+        /// environment.
+        /// </summary>
+        /// <param name="stackName">The name of the stack.</param>
+        /// <param name="environments">List of environments to add to the end of the stack's import list.</param>
+        /// <param name="cancellationToken">A cancellation token.</param>
+        public abstract Task AddEnvironmentsAsync(string stackName, IEnumerable<string> environments, CancellationToken cancellationToken = default);
```

Comment:
The new APIs in the other SDKs use varargs for the `environments` arg ([Node](https://github.com/pulumi/pulumi/blob/6ae7c39498cc476d9240d93d52c9bd79cfcafd4b/sdk/nodejs/automation/workspace.ts#L102), [Python](https://github.com/pulumi/pulumi/blob/6ae7c39498cc476d9240d93d52c9bd79cfcafd4b/sdk/python/lib/pulumi/automation/_workspace.py#L210), and [Go](https://github.com/pulumi/pulumi/blob/6ae7c39498cc476d9240d93d52c9bd79cfcafd4b/sdk/go/auto/workspace.go#L48)):

```typescript
// Usage: await addEnvironments("stackname", "environment1", "environment2");
addEnvironments(stackName: string, ...environments: string[]): Promise<void>;
```

```python
# Usage: add_environment("stackname", "environment1", "environment2")
def add_environments(self, stack_name: str, *environment_names: str) -> None:
```

```go
// Usage: AddEnvironment(ctx, "stackname", "environment1", "environment2")
AddEnvironments(context.Context, string, ...string) error
```

We could add an overload that accepted a `params` array to make it similarly more ergonomic to use in most cases when you don't care about passing a cancellation token:

```c#
        /// <summary>
        /// Adds environments to the end of a stack's import list. Imported environments are merged in order
        /// per the ESC merge rules. The list of environments behaves as if it were the import list in an anonymous
        /// environment.
        /// </summary>
        /// <param name="stackName">The name of the stack.</param>
        /// <param name="environments">List of environments to add to the end of the stack's import list.</param>
        public Task AddEnvironmentsAsync(string stackName, params string[] environments)
            => AddEnvironmentsAsync(stackName, (IEnumerable<string>)environments);
```

Then usage in most cases would look like:

```c#
await workspace.AddEnvironmentsAsync(stackName, "automation-api-test-env", "automation-api-test-env-2");
```

Rather than:

```c#
await workspace.AddEnvironmentsAsync(stackName, new[] { "automation-api-test-env", "automation-api-test-env-2" });
```

And if we add it here, add one to `WorkspaceStack` as well.

Aside: We could likely do that for other cases where we accept an `IEnumerable<string>` arg.

Reply:
@justinvp I agree with all of this and will follow up. Any reservation on that coming in a follow up PR and merging this PR?

Reply:
Follow-up PR works for me!